kernel/kernel-benchmarks/src/test/java/io/delta/kernel/benchmarks/workloadrunners/WorkloadRunner.java

```diff
@@ -72,6 +84,45 @@ public WorkloadRunner() {}
    */
   public abstract void cleanup() throws Exception;
 
+  /**
+   * Loads a snapshot for the table.
+   *
+   * <p>For CatalogManaged tables, uses {@link UCCatalogManagedClient} to handle staged commits. For
+   * regular tables, uses {@link TableManager}.
+   *
+   * @param engine the engine to use
+   * @param tableInfo the table information
+   * @param versionOpt optional version to load (if empty, loads latest)
+   * @return a Snapshot for the table
+   * @throws Exception if there's an error loading the snapshot
+   */
+  protected Snapshot loadSnapshot(Engine engine, TableInfo tableInfo, Optional<Long> versionOpt)
+      throws Exception {
+    String tableRoot = tableInfo.getResolvedTableRoot();
+
+    if (tableInfo.isCatalogManaged()) {
+      CatalogManagedInfo catalogManagedInfo = tableInfo.getCatalogManagedInfo();
+      InMemoryUCClient ucClient = catalogManagedInfo.createUCClient(engine, tableRoot);
+      UCCatalogManagedClient catalogClient = new UCCatalogManagedClient(ucClient);
```

Comment:
`ucCatalogManagedClient` -- `ucClient` and `catalogClient` are a bit too similar

And yes -- there's to many "uc clients" -- that's not your fault :p